### PR TITLE
Revamp UI DG, Remove unused cards and tables

### LINKED ISSUE #157: [REFACTOR] Remove unused artifacts and class methods from codebase
# Summary

## Check through and clean up codebase by:
1. Removing unused classes and files
2. Removing unused methods in classes

### LINKED ISSUE #195: [DOC] Simon Write at least 3 pages worth of DG in total

### LINKED ISSUE #206: [DOC] Clean Up Design and Implementation Artefact from AB3 DG UI

### PR DESCRIPTION
**Pull request by simonjulianl**
### Summary

* Resolves #195 
* Resolves #206 
* Resolves #157 
* Also make enhancement to UG by reordering it

### Detail

* Reorder some of the UGs section to make it look logical 
* Update the UI component section by enhancing the current diagram and creating 1 more sequence diagram 
* Also delete the unused cards and tables that are never deleted 

### Checks

- [x] Link PR to issue
- [x] Add Reviewer
- [x] Pass CI

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -87,7 +87,7 @@ in [`Ui.java`](https://github.com/AY2122S1-CS2103T-T15-1/tp/tree/master/src/main
 
 ![Structure of the UI Component](images/UiClassDiagram.png)
 
-The UI consists of a `MainWindow` that is made up of parts e.g.`CommandBox`, `ResultDisplay`, `PersonListPanel`
+The UI consists of a `MainWindow` that is made up of many Ui components e.g.`CommandBox`, `ResultDisplay`, `PersonListPanel`
```
Comment (radiankrisno) [inline docs/DeveloperGuide.md:90]: I think Ui no longer use PersonListPanel

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -375,14 +384,7 @@ The following activity diagram summarizes what happens when a user executes a ne
 * **Alternative 2:** Individual command knows how to undo/redo by itself.
     * Pros: Will use less memory (e.g. for `delete`, just save the person being deleted).
```
Comment (radiankrisno) [inline docs/DeveloperGuide.md:385]: no longer have person

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -650,46 +652,35 @@ testers are expected to do more *exploratory* testing.
...
 
-    1. Other incorrect delete commands to try: `delete`, `delete x`, `...` (where x is larger than the list size)<br>
-       Expected: Similar to previous.
-
-1. _{ more test cases …​ }_
+    4. Other incorrect delete commands to try: `delete t/activity`, `delete t/doctor`, `delete t/activity x`, `...` (where x is an invalid id)<br>
+       Expected: Similar to previous for each patient, doctor and activity model.
 
 ### Finding a patient, doctor or activity
 1. Searching for a person
```
Comment (radiankrisno) [inline docs/DeveloperGuide.md:674]: delete this maybe?

**Comment by simonjulianl** [comment]
Fixed @radiankrisno

**Review by radiankrisno — APPROVED** [review]
LGTM

**Pull request merged**